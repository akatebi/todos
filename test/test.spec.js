import {
  AddUserTest,
  RemoveUserTest,
  AddTodoTest,
  QueryUserTest,
  MarkAllTodosTest,
  ClearCompletedTodosTest
} from "./schema";

describe("Remove User", RemoveUserTest("user1"));
describe("Add User", AddUserTest("user1"));
describe("Add Todo", AddTodoTest("Get A Customer"));
describe("Query User", QueryUserTest("user1"));
describe("Mark All Todos", MarkAllTodosTest());
describe("Clear Completed Todos", ClearCompletedTodosTest());
