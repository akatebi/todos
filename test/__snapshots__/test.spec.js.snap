// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Todo  1`] = `
Object {
  "data": Object {
    "addTodo": Object {
      "clientMutationId": "AddToDo-0",
      "todoEdge": Object {
        "cursor": "NTg=",
        "node": Object {
          "complete": false,
          "id": "VG9kbzo1OA==",
          "text": "Get A Customer 1",
        },
      },
      "user": Object {
        "completedCount": 0,
        "email": "user1@test.com",
        "totalCount": 1,
      },
    },
  },
}
`;

exports[`Add Todo  2`] = `
Object {
  "data": Object {
    "addTodo": Object {
      "clientMutationId": "AddToDo-1",
      "todoEdge": Object {
        "cursor": "NTk=",
        "node": Object {
          "complete": false,
          "id": "VG9kbzo1OQ==",
          "text": "Get A Customer 2",
        },
      },
      "user": Object {
        "completedCount": 0,
        "email": "user1@test.com",
        "totalCount": 2,
      },
    },
  },
}
`;

exports[`Add Todo  3`] = `
Object {
  "data": Object {
    "addTodo": Object {
      "clientMutationId": "AddToDo-2",
      "todoEdge": Object {
        "cursor": "NjA=",
        "node": Object {
          "complete": false,
          "id": "VG9kbzo2MA==",
          "text": "Get A Customer 3",
        },
      },
      "user": Object {
        "completedCount": 0,
        "email": "user1@test.com",
        "totalCount": 3,
      },
    },
  },
}
`;

exports[`Add User  1`] = `
Object {
  "data": Object {
    "addUser": Object {
      "clientMutationId": "user1",
      "id": "VXNlcjoyOQ==",
    },
  },
}
`;

exports[`Change Todo Status => false  1`] = `
Object {
  "data": Object {
    "changeTodoStatus": Object {
      "clientMutationId": "ChangeTodoStatus-0",
      "todo": Object {
        "complete": false,
        "id": "VG9kbzo1OA==",
        "text": "Get A Customer 1",
      },
      "user": Object {
        "completedCount": 2,
        "email": "user1@test.com",
        "id": "VXNlcjoyOQ==",
        "totalCount": 3,
      },
    },
  },
}
`;

exports[`Change Todo Status => false  2`] = `
Object {
  "data": Object {
    "changeTodoStatus": Object {
      "clientMutationId": "ChangeTodoStatus-1",
      "todo": Object {
        "complete": false,
        "id": "VG9kbzo1OQ==",
        "text": "Get A Customer 2",
      },
      "user": Object {
        "completedCount": 1,
        "email": "user1@test.com",
        "id": "VXNlcjoyOQ==",
        "totalCount": 3,
      },
    },
  },
}
`;

exports[`Change Todo Status => false  3`] = `
Object {
  "data": Object {
    "changeTodoStatus": Object {
      "clientMutationId": "ChangeTodoStatus-2",
      "todo": Object {
        "complete": false,
        "id": "VG9kbzo2MA==",
        "text": "Get A Customer 3",
      },
      "user": Object {
        "completedCount": 0,
        "email": "user1@test.com",
        "id": "VXNlcjoyOQ==",
        "totalCount": 3,
      },
    },
  },
}
`;

exports[`Change Todo Status => true  1`] = `
Object {
  "data": Object {
    "changeTodoStatus": Object {
      "clientMutationId": "ChangeTodoStatus-0",
      "todo": Object {
        "complete": true,
        "id": "VG9kbzo1OA==",
        "text": "Get A Customer 1",
      },
      "user": Object {
        "completedCount": 1,
        "email": "user1@test.com",
        "id": "VXNlcjoyOQ==",
        "totalCount": 3,
      },
    },
  },
}
`;

exports[`Change Todo Status => true  2`] = `
Object {
  "data": Object {
    "changeTodoStatus": Object {
      "clientMutationId": "ChangeTodoStatus-1",
      "todo": Object {
        "complete": true,
        "id": "VG9kbzo1OQ==",
        "text": "Get A Customer 2",
      },
      "user": Object {
        "completedCount": 2,
        "email": "user1@test.com",
        "id": "VXNlcjoyOQ==",
        "totalCount": 3,
      },
    },
  },
}
`;

exports[`Change Todo Status => true  3`] = `
Object {
  "data": Object {
    "changeTodoStatus": Object {
      "clientMutationId": "ChangeTodoStatus-2",
      "todo": Object {
        "complete": true,
        "id": "VG9kbzo2MA==",
        "text": "Get A Customer 3",
      },
      "user": Object {
        "completedCount": 3,
        "email": "user1@test.com",
        "id": "VXNlcjoyOQ==",
        "totalCount": 3,
      },
    },
  },
}
`;

exports[`Clear Completed Todos  1`] = `
Object {
  "data": Object {
    "clearCompletedTodos": Object {
      "clientMutationId": "ClearCompletedTodos",
      "deletedTodoIds": Array [
        "VG9kbzo1OA==",
        "VG9kbzo1OQ==",
        "VG9kbzo2MA==",
      ],
      "user": Object {
        "completedCount": 0,
        "id": "VXNlcjoyOQ==",
        "totalCount": 0,
      },
    },
  },
}
`;

exports[`Mark All Todos  1`] = `
Object {
  "data": Object {
    "markAllTodos": Object {
      "changedTodos": Array [
        Object {
          "complete": true,
          "id": "VG9kbzo1OA==",
          "text": "Get A Customer 1",
        },
        Object {
          "complete": true,
          "id": "VG9kbzo1OQ==",
          "text": "Get A Customer 2",
        },
        Object {
          "complete": true,
          "id": "VG9kbzo2MA==",
          "text": "Get A Customer 3",
        },
      ],
      "clientMutationId": "MarkAllTodos",
      "user": Object {
        "completedCount": 3,
        "id": "VXNlcjoyOQ==",
        "totalCount": 3,
      },
    },
  },
}
`;

exports[`Query User  1`] = `
Object {
  "data": Object {
    "user": Object {
      "completedCount": 0,
      "email": "user1@test.com",
      "id": "VXNlcjoyOQ==",
      "todos": Object {
        "edges": Array [
          Object {
            "cursor": "NTg=",
            "node": Object {
              "__typename": "Todo",
              "complete": false,
              "id": "VG9kbzo1OA==",
            },
          },
          Object {
            "cursor": "NTk=",
            "node": Object {
              "__typename": "Todo",
              "complete": false,
              "id": "VG9kbzo1OQ==",
            },
          },
          Object {
            "cursor": "NjA=",
            "node": Object {
              "__typename": "Todo",
              "complete": false,
              "id": "VG9kbzo2MA==",
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "NjA=",
          "hasNextPage": true,
        },
      },
      "totalCount": 3,
    },
  },
}
`;

exports[`Remove User  1`] = `
Object {
  "data": Object {
    "removeUser": Object {
      "clientMutationId": "user1",
    },
  },
}
`;
